src/vector/v.surf.icw/v.surf.icw.py

```diff
@@ -534,23 +534,31 @@ def main():
     # grass.run_command('g.list', type = 'raster', mapset = '.')
 
     #######################################################
-    grass.message("")
+    grass.verbose("\n")
     grass.message(_("Calculating final values ..."))
 
-    input_maps = tmp_base + "partial.%05d" % 1
-    for i in range(2, n + 1):
-        input_maps += ",%spartial.%05d" % (tmp_base, i)
+    TMP_FILE = grass.tempfile()
```

Comment:
See https://github.com/OSGeo/grass/blob/main/doc/development/style_guide.md#temporary-files

g.tempfile is meant for large data, in this case standard system tempfile seems better.

Reply:
See also how to deal with temporary maps in the guide. Right now you call cleanup function explicitly, but does it get called again since it's registered with atexit?

Reply:
> See also how to deal with temporary maps in the guide. Right now you
> call cleanup function explicitly, but does it get called again since it's
> registered with atexit?

It gets called on exit a second time, but only tries to remove if something actually exists. It could be improved, tmp_base should be a global variable for one thing.

Reply:
> See https://github.com/OSGeo/grass/blob/main/doc/development/style_guide.md#temporary-files
> 
> g.tempfile is meant for large data, in this case standard system tempfile seems better.

Any particular reason why? I don't really mind either way but g.tempfile only takes 1.36 milliseconds to complete for me, is only run twice during a long running module, doesn't pollute the filesystem outside of the mapset's .tmp dir, and gets a second chance to be cleared when the GRASS session ends/launches if the original core.try_remove() fails for some reason.   g.tempfile just tries to create the file and reports back what its name is, it doesn't know or care about what ends up in it.   For sure large files should use g.tempfile so that the disk space is used in MAPSET and we don't risk filling up a small /tmp partition.

Reply:
Yes, g.tempfile is meant for large files and also for Bash/shells, but otherwise there is little reason to use it. 1) In general, relying on a standard library, in this case Python standard library, is better than relying on our custom mechanism. The library provides standardized, well-document and well-tested tools with known limits. 2) Security concerns are known and addressed. 3) Code analyzers can understand the code better. 4) Other readers need to know only standard Python, not GRASS GIS specific API. 5) Creating files in the mapset can be significantly slower than creating them in system's tmp dir. That dir is meant to be fast, that's at least what most application would expect. /tmp is sometimes mounted to memory. On the other hand, mapset dir may be on a slow network drive for storage of GRASS projects, large data, or because of parallel, multi-node processing in the same GRASS project.

Thinking about this more, some Python functions for tmp files and dirs allow you to pick the base tmp dir, so you can use the standard Python functions with mapset's .tmp if large files are a concern. (I used a custom dir approach to implement a temporary mapset session ([TemporaryMapsetSession](https://github.com/OSGeo/grass/blob/main/python/grass/experimental/mapset.py#L120), [create_temporary_mapset](https://github.com/OSGeo/grass/blob/main/python/grass/experimental/create.py#L56)).) If that seems like a good way, we may want to document that g.tempfile is meant only for Bash/shells scripts and perhaps add a convenient wrapper for Python functions to work over mapset's tmp dir.

Another take on the situation is that the standard Python functions (and related code quality checks) are designed in a way that the code should be sure everything is properly deleted at the right time. The `with` statement is the prime example of that and that's what is usually appropriate and sufficient in individual scripts or tools (it is more difficult in the library where things should to be designed to work with the `with` statement, but may not be able use it themselves). This is different from the older approach of deleting on couple different levels or places in case the previous delete didn't work out or was forgotten.